src/aequitas/bias.py

```diff
@@ -184,16 +324,335 @@ def get_disparity_predefined_groups(self, df, ref_groups_dict, key_columns=None,
+                difference_significance_p
+
+        return disparity_df
+
+
+
+    @classmethod
+    def get_statistical_significance(cls, original_df, disparity_df, ref_dict,
+                                     score_thresholds=None, model_id=1,
+                                     attr_cols=None, aplha=5e-2):
+        '''
+
+        :param original_df: a dataframe containing a required raw 'score' column
+            and possible raw 'label_value' column.
+        :param disparity_df: Expansion of get_crosstabs() output with additional
+            columns for disparity metrics and each metric's reference group to
+            which significance must be added
+        :param ref_dict: Dictionary indicating reference group for each
+            attribute/ measure combination
+        :param score_thresholds: a dictionary { 'rank_abs':[] , 'rank_pct':[], 'score':[] }
+        :param model_id: (Future functionality) ID(s) of models for which to check
+            statistical significance
+        :param attr_cols: Columns indicating attribute values in original_df
+        :param aplha: Level at which to determine statistical significance.
+            Default is 5e-2 (0.05).
+
+        :return: dataframe with same number of rows as the input (crosstab)
+            but with additional disparity metrics columns, ref_group_values, and
+            statisitcal significance (p-values) of specific metrics.
+        '''
+        if 'label_value' not in original_df.columns:
+            raise ValueError(
+                "Column 'label_value' not in dataframe. Label values are "
+                "required for computing statistical significance of supervised "
+                "metrics.")
+
+        if attr_cols is None:
+            non_attr_cols = [
+                'id', 'model_id', 'entity_id', 'score', 'label_value',
+                'rank_abs', 'rank_pct']
+            # index of the columns that are attributes
+            attr_cols = original_df.columns[~original_df.columns.isin(non_attr_cols)]
+
+        for col in attr_cols:
+            # find the priors_df
+            col_group = original_df.fillna({col: 'pd.np.nan'}).groupby(col)
+            counts = col_group.size()
+
+        # check if all attr_cols exist in df
+        check = [col in original_df.columns for col in attr_cols]
+        if False in check:
+            raise ValueError(
+                f"Not all attribute columns provided '{attr_cols}' exist in "
+                f"input dataframe!")
+
+        # check if all columns are strings:
+        non_string_cols = \
+            original_df.columns[(original_df.dtypes != object) & (original_df.dtypes != str) & (
+                original_df.columns.isin(attr_cols))]
+
+        if non_string_cols.empty is False:
+            logging.error(
+                'get_statistical_significance: statistical significance was '
+                'not calculated. There are non-string cols within attr_cols.')
+            exit(1)
```

Comment:
😿  exit?!

That seems extreme outside of an explicitly-CLI / process-management context. Do you just want to raise an exception? Or return?

I understand that perhaps you really want to `exit(1)`, even ... but that seems out of place at this layer.

Reply:
@jesteria this chunk of get_statistical_significance() method is actually modeled after existing code for similar grouping and thresholding tasks in the Group() class. 

I included the exit b/c I thought that was what DSaPP team decided was appropriate when the original df (vs. crosstab or any of the dfs calculated with the library) doesn't follow expected format. Not sure if you were involved with Aequitas at the time it was written, but if that exit isn't a requirement, happy to remove. Would also love to chat briefly about what is more Pythonic re: exception or return here!

Reply:
well aequitas became a library later, but now it makes sense to remove exits from core src/aequitas and the exists should be triggered when the exceptions are caught on the aequitas_cli. I've experienced this awkward behaviour recently when using aequitas with a customized df (which was not an output of get_crosstabs) and jupyter was printing exits :)

Reply:
Removed this, thanks both!